api/src/main/java/com/graphhopper/config/LMProfileConfig.java

```diff
@@ -47,17 +48,35 @@ void setProfile(String profile) {
         this.profile = profile;
     }
 
+    public boolean usesOtherPreparation() {
+        return !preparationProfile.equals("self");
+    }
+
+    public String getPreparationProfile() {
+        return preparationProfile;
+    }
+
+    public LMProfileConfig setPreparationProfile(String preparationProfile) {
+        validateProfileName(preparationProfile);
+        if (maximumLMWeight >= 0)
+            throw new IllegalArgumentException("Using non-default maximum_lm_weight and preparation_profile at the same time is not allowed");
```

Comment:
Yeah you mean the hints need to be emtpy as well?

Reply:
Yes (and could be even checked earlier after parsing?)

Reply:
This is kind of checked during parsing because Jackson uses this setter. And I just noticed there aren't any other 'hints', so checking the maxLMWeight should be sufficient here.